src/verify.rs

```diff
@@ -89,15 +89,35 @@ use self::PreparedStep::*;
 ///
 /// This type would be Copy except for the fact that the bitset can require
 /// overflow storage :(.
-struct StackSlot {
+#[derive(Clone)]
+pub struct StackSlot {
     vars: Bitset,
     code: Atom,
     expr: Range<usize>,
 }
 
+/// A constructor trait for plugging in to the verifier, to collect extra data during the
+/// verification pass
+pub trait ProofBuilder {
```

Comment:
The more I think about it, I agree with this — as bad as it is now, it'll be worse once we have explicit and packed support.  Better to have one proof parser.